### DISCUSSION TITLE
Hello and a few questions

### DISCUSSION THREAD

**Discussion by paulhuggett** [midi2-dev/AM_MIDI2.0Lib#12] [General]
Hello! I was starting to learn about MIDI 2.0 with a view to creating an implementation for a little project I'm working on. I came across your library and it looks to be similar to what I was intending to write. I'm happy to contribute to its development where I can rather than reinvent the wheel!

A few questions for you:

- The project's readme says "The code is still being adjusted as it is being prototyped and changes do occur, however the WIKI attempts to reflect the latest code." Which wiki is it referring to?
- Would you be open to using clang-format to create a uniform code layout? The chosen formatting  doesn't bother me, but I'm finding the mixed use of tabs and spaces hard to deal with!
- Perhaps using github actions to setup some sort of continuous integration and/or static analysis? (Maybe it's too early in the library's life to be setting that up but perhaps the sonner the better!)

Assuming you're happy, I'm sure I'll have more questions and PRs as I get into it!

**Comment by starfishmod** [comment]
@paulhuggett hey thanks for reaching out glad this project does most of what you wanted :)

> The project's readme says "The code is still being adjusted as it is being prototyped and changes do occur, however the WIKI attempts to reflect the latest code." Which wiki is it referring to?

Yeah the "docs" folder is what I meant. I did have this as WIKI when I had this is a private host. Now that it is here I would like to get all that moved into a GH WIKI. - Just enabled the WIKI so I'll move thhe docs there.

> Would you be open to using clang-format to create a uniform code layout? The chosen formatting doesn't bother me, but I'm finding the mixed use of tabs and spaces hard to deal with!

Sure... the tabs and spaces thing is due to different IDE's... It doesn't bother me but I see why it bothers everyone else :laughing: 

> Perhaps using github actions to setup some sort of continuous integration and/or static analysis? (Maybe it's too early in the library's life to be setting that up but perhaps the sonner the better!)

Good idea - It's just a matter of time to do the setup!

>Assuming you're happy, I'm sure I'll have more questions and PRs as I get into it!

Absolutely - Everything is Welcome! I'm just starting to go through all your PR's - please bear with me as I review.

**Reply by paulhuggett** [reply]
> Good idea - It's just a matter of time to do the setup!

FWIW, I've been continuing to modify the library over the past month or so. My focus has mostly been on squashing compiler warnings (when they are cranked up), improving test coverage using [Google Test](https://github.com/google/googletest) and adding some fuzz testing courtesy of [Google FuzzTest](https://github.com/google/fuzztest). 

I've added a [.clang-format configuration file](https://github.com/paulhuggett/AM_MIDI2.0Lib/blob/main/.clang-format) and GitHub Actions for continuous integration, code coverage analysis, and fuzzing. The workflows are mostly imported directly from [another project](https://github.com/paulhuggett/icubaby/); I'll gradually bring some of the others across as well.

As always, if you think that any of my changes belong in the original library, I'm more than happy to submit PRs.
